cscs-checks/microbenchmarks/mpi/mpi_helloworld.py

```diff
@@ -0,0 +1,36 @@
+import os
+import reframe.utility.sanity as sn
+
+from reframe.core.deferrable import deferrable
+from reframe.core.pipeline import RegressionTest
+
+
+class MPIHelloWorldTest(RegressionTest):
+    def __init__(self, **kwargs):
+        super().__init__('mpi_helloworld_check',
+                         os.path.dirname(__file__), **kwargs)
+        self.valid_systems = ['daint:gpu', 'daint:mc',
+                              'dom:gpu', 'dom:mc',
+                              'kech:cn', 'kech:pn',
+                              'leone:normal', 'monch:compute']
+        self.descr = 'MPI Hello World'
+        self.sourcepath = 'mpi_helloworld.c'
+        self.executable = './mpi_helloworld'
+        self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu',
+                                    'PrgEnv-intel', 'PrgEnv-pgi']
+        self.maintainers = ['RS', 'VK']
+        self.num_tasks_per_node = 1
+        self.num_tasks = 0
+        num_processes = sn.extractsingle(
+            r'Received messages from (?P<nprocs>\d+) processes',
+            self.stdout, 'nprocs', int)
+        self.sanity_patterns = sn.assert_eq(num_processes, self.real_num_tasks)
+
+    @property
+    @deferrable
```

Comment:
The `@deferrable` decorator is for internal use. You should `@sn.sanity_function` instead.